node_test.go

```diff
@@ -1142,18 +1046,15 @@ func TestNodeCommitPaginationAfterRestart(t *testing.T) {
 	cfg.MaxSizePerMsg = size - uint64(s.ents[len(s.ents)-1].Size()) - 1
 
 	rn, err := NewRawNode(cfg)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
+
 	n := newNode(rn)
 	go n.run()
 	defer n.Stop()
 
 	rd := readyWithTimeout(&n)
-	if !IsEmptyHardState(rd.HardState) && rd.HardState.Commit < persistedHardState.Commit {
-		t.Errorf("HardState regressed: Commit %d -> %d\nCommitting:\n%+v",
-			persistedHardState.Commit, rd.HardState.Commit,
-			DescribeEntries(rd.CommittedEntries, func(data []byte) string { return fmt.Sprintf("%q", data) }),
-		)
-	}
+	assert.False(t, !IsEmptyHardState(rd.HardState) && rd.HardState.Commit < persistedHardState.Commit,
```

Comment:
nit
```suggestion
	assert.True(t, IsEmptyHardState(rd.HardState) && rd.HardState.Commit < persistedHardState.Commit,
```

Reply:
`IsEmptyHardState(rd.HardState) || rd.HardState.Commit >= persistedHardState.Commit`?

Reply:
Good catch. I did not see the `&&`.

Reply:
Overall not a big deal, let's keep it as it's for now. Please feel free to update it in a followup PR if you want.